osquery/remote/http/http_client.cpp

```diff
@@ -0,0 +1,342 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/http_client.h>
+#include <osquery/logger.h>
+
+namespace osquery {
+namespace http {
+
+/** This class is used to convert boost::system_exception
+ *  to std::system_exception, since on freebsd osquery is on boost-1.64.
+ *  Idea taken from boost-1.65
+ */
+class adapted_category : public std::error_category {
+ public:
+  explicit adapted_category(boost::system::error_category const* pc)
+      : pc_(pc) {}
+
+  const char* name() const noexcept {
+    return pc_->name();
+  }
+
+  std::string message(int ev) const {
+    return pc_->message(ev);
+  }
+
+ private:
+  boost::system::error_category const* pc_;
+};
+
+/** In the postResponseHandler, treating SHORT_READ_ERROR as success
+ *  for ssl connections. This can happen if a remote server did not
+ *  call shutdown on ssl connection.
+ */
+void Client::postResponseHandler(boost_system::error_code const& ec) {
+  if ((ec.category() == boost_asio::error::ssl_category) &&
+      (ec.value() == SHORT_READ_ERROR)) {
+    // ignoring short read error
+    ec_ = boost_system::errc::make_error_code(boost_system::errc::success);
+  } else if ((ec.value() != boost_system::errc::operation_canceled) ||
+             (ec.category() != boost_asio::error::system_category)) {
```

Comment:
You should never have to call `value()` or `category()` under normal usage, the following should work:
```
} else if (ec != boost_system::errc::operation_canceled) {
```